codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/RestXmlResponseProtocolGenerator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.typescript.codegen;
+
+import java.util.List;
+import java.util.Set;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.codegen.core.SymbolReference;
+import software.amazon.smithy.model.knowledge.HttpBinding;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.HttpBindingProtocolGenerator;
+
+/**
+ * Handles general components across the XML-document response based AWS protocols
+ * that have HTTP bindings. It handles reading from document bodies, including
+ * generating any functions needed for performing deserialization.
+ *
+ * This builds on the foundations of the {@link HttpBindingProtocolGenerator} to
+ * handle components of binding to HTTP responses.
+ *
+ * @see XmlShapeDeserVisitor
+ * @see XmlMemberDeserVisitor
+ * @see AwsProtocolUtils
+ * @see <a href="https://awslabs.github.io/smithy/spec/http.html">Smithy HTTP protocol bindings.</a>
+ * @see <a href="https://awslabs.github.io/smithy/spec/xml.html">Smithy XML traits.</a>
+ */
+abstract class RestXmlResponseProtocolGenerator extends HttpBindingProtocolGenerator {
```

Comment:
I don't think this level of abstraction is needed. Let's just roll this into AwsRestXml.